src/Component.php

```diff
@@ -5,12 +5,155 @@
+        }
+    }
+
+    private function writeManifest(SplFileInfo $file, array $manifest, array $header) : void
+    {
+        $manifest["columns"][] = $header;
+        $targetManifest = $this->getDataDir() . '/out/tables/' . $file->getFilename() . ".manifest";
+        file_put_contents(
+            $targetManifest,
+            $this->jsonEncode->encode($manifest, JsonEncoder::FORMAT)
+        );
+    }
+
+    private function readManifest(SplFileInfo $file) : array
+    {
+        $manifestFile = $file->getPathname() . '.manifest';
+        if (!$this->fs->exists($manifestFile)) {
+            throw new UserException(
+                'Table ' . $file->getBasename() . ' does not have a manifest file.'
+            );
+        }
+
+        $manifest = $this->jsonDecode->decode(file_get_contents($manifestFile), JsonEncoder::FORMAT);
+        $this->validateManifest($manifest, $file->getFilename());
+        return $manifest;
+    }
+
     public function run(): void
     {
-        // @TODO implement
+        $finder = new Finder();
+        $finder->in($this->getDataDir() . '/in/tables')->notName('*.manifest')->depth(0);
+        foreach ($finder as $file) {
+            $manifest = $this->readManifest($file);
+            $header = $manifest['columns'];
+            $delimiter = $manifest['delimiter'];
+            $enclosure = $manifest['enclosure'];
+
+            if (is_dir($file->getPathname())) {
+                // sliced file
+                $finder = new Finder();
+                $finder->in($file->getPathname())->files();
+                $slicedDestination = $this->getDataDir() . '/out/tables/' . $file->getFilename();
+                $this->fs->mkdir($slicedDestination);
+                $maxColCount = 0;
+                foreach ($finder as $slicedFile) {
+                    $maxColCount = max(
+                        $maxColCount,
+                        $this->getMaxColCount($slicedFile, $delimiter, $enclosure)
+                    );
+                }
+                $header = $this->fillHeader($header, $maxColCount);
+                foreach ($finder as $slicedFile) {
+                    $destination = $this->getDataDir() . '/out/tables/' .
+                        $file->getFilename() . '/' . $slicedFile->getFilename();
+                    $this->orthogonalize($slicedFile, $destination, $maxColCount, $delimiter, $enclosure);
+                }
+            } else {
+                $maxColCount = $this->getMaxColCount($file, $delimiter, $enclosure);
+                $header = $this->fillHeader($header, $maxColCount);
+                $destination = $this->getDataDir() . '/out/tables/' . $file->getFilename();
+                $this->orthogonalize($file, $destination, $maxColCount, $delimiter, $enclosure);
+            }
+
+            $this->writeManifest($file, $manifest, $header);
+        }
+    }
+
+    public function __construct()
+    {
+        parent::__construct();
```

Comment:
Tohle vypadá jako fajn způsob, jak to dělat. 👍

Reply:
co presne myslis?

Reply:
Inicializace těch dependencies v konstruktoru...